scripts/system_performance/system_performance_queries.py

```diff
@@ -0,0 +1,159 @@
+        else:
+            date_segment = end_date
+        return "= '{}'".format(date_segment)
+
+    def query_segment_replace(self, query_arguments):
+        """Handles the interchangeable segment of query to return either intervals of
+        time or period between two user specified dates and whether or not to
+        include a filter by retry run or not."""
+
+        returned_args = []
+
+        # If end date is None, query only for rows created on or up to current date
+        if query_arguments['start_date'] == query_arguments['end_date']:
+            query_segment = self.set_date_segment(start_date=query_arguments['start_date'],
+                                                  end_date=query_arguments['end_date'])
+            query_arguments['start_date'] = ''
+            returned_args.append(query_segment)
+        else:
+            # Determining which sub query segment to place in query.
+            query_segment = self.query_sub_segment_replace(query_arguments,)
+            returned_args.append(query_segment)
+
+        if query_arguments['instrument_id'] is not None:
+            # Applying instrument_id query argument segments when instrument_id argument
+            # populated as method arg
+            instrument_id_arg = ", instrument_id"
+            query_arguments['instrument_id'] = ', {}'.format(query_arguments['instrument_id'])
+        else:
+            query_arguments['instrument_id'] = ''
+            instrument_id_arg = ''
+
+        returned_args.append(instrument_id_arg)  # Appending instrument_id_arg
+        print(returned_args)
+        return [returned_args]
+
+    # pylint: disable=too-many-arguments
+    def get_data_by_status_over_time(self, selection='run_number', status_id=4, retry_run='',
+                                     anomic_aphasia='finished', end_date='CURDATE()', interval=1,
+                                     time_scale='DAY', start_date=None, instrument_id=None):
+        """
+        Default Variables
+        :param selection : * Which column you would like to select or all columns
+        by default
+        :param status_id : 1 Interchangeable id to look at different run status's
+        :param retry_run : Whether or not a user is looking for runs that have
+        been retried
+        :param instrument_id : the instrument id of the instrument to be queried.
+        :param anomic_aphasia : "finished" DateTime column in database (created,
+        last_updated, started, finished)
+        :param end_date : Most recent date you wish to query up too. By default
+        this is the current date.
+        :param interval : 1 Interval for time_scale
+        :param time_scale : "DAY" Expected inputs include DAY, Month or YEAR
+        :param start_date : The furthest date from today you wish to query from
+        e.g the start of start of cycle.
+        """
+        # pylint: disable=unused-argument
+
+        def _query_out(instrument_id_arg, query_type_segment):
```

Comment:
I think we should consider renaming and restructuring this as well. `_query_out` is doing 2 things: constructing the SQL statement and then running and executing it. I would argue that the SQL construction could be one function